src/index.js

```diff
@@ -1,407 +1,11 @@
-	return vals ? vals.slice() : [];
-};
-
-Entity.prototype.getLinkByClasses = function(linkClasses) {
-	const vals = getMatchingValuesByAll(this.links, linkClasses, 'class');
-	return vals && vals.length > 0 ? vals[0] : undefined;
-};
-
-Entity.prototype.getLinksByClasses = function(linkClasses) {
-	const vals = getMatchingValuesByAll(this.links, linkClasses, 'class');
-	return vals && vals.length > 0 ? vals.slice() : [];
-};
-
-Entity.prototype.getLinkByType = function(linkType) {
-	const vals = getMatchingValue(this._linksByType, linkType);
-	return vals ? vals[0] : undefined;
-};
-
-Entity.prototype.getLinksByType = function(linkType) {
-	const vals = getMatchingValue(this._linksByType, linkType);
-	return vals ? vals.slice() : [];
-};
-
-Entity.prototype.getSubEntity = function(entityRel) {
-	return this.getSubEntityByRel(entityRel);
-};
-
-Entity.prototype.getSubEntities = function(entityRel) {
-	return this.getSubEntitiesByRel(entityRel);
-};
-
-Entity.prototype.getSubEntityByRel = function(entityRel) {
-	const vals = getMatchingValue(this._entitiesByRel, entityRel);
-	return vals ? vals[0] : undefined;
-};
-
-Entity.prototype.getSubEntitiesByRel = function(entityRel) {
-	const vals = getMatchingValue(this._entitiesByRel, entityRel);
-	return vals ? vals.slice() : [];
-};
-
-Entity.prototype.getSubEntityByRels = function(entityRels) {
-	const vals = getMatchingValuesByAll(this.entities, entityRels, 'rel');
-	return vals && vals.length > 0 ? vals[0] : undefined;
-};
-
-Entity.prototype.getSubEntitiesByRels = function(entityRels) {
-	const vals = getMatchingValuesByAll(this.entities, entityRels, 'rel');
-	return vals && vals.length > 0 ? vals.slice() : [];
-};
-
-Entity.prototype.getSubEntityByClass = function(entityClass) {
-	const vals = getMatchingValue(this._entitiesByClass, entityClass);
-	return vals ? vals[0] : undefined;
-};
-
-Entity.prototype.getSubEntitiesByClass = function(entityClass) {
-	const vals = getMatchingValue(this._entitiesByClass, entityClass);
-	return vals ? vals.slice() : [];
-};
-
-Entity.prototype.getSubEntityByClasses = function(entityClasses) {
-	const vals = getMatchingValuesByAll(this.entities, entityClasses, 'class');
-	return vals && vals.length > 0 ? vals[0] : undefined;
-};
-
-Entity.prototype.getSubEntitiesByClasses = function(entityClasses) {
-	const vals = getMatchingValuesByAll(this.entities, entityClasses, 'class');
-	return vals && vals.length > 0 ? vals.slice() : [];
-};
-
-Entity.prototype.getSubEntityByType = function(entityType) {
-	const vals = getMatchingValue(this._entitiesByType, entityType);
-	return vals ? vals[0] : undefined;
-};
-
-Entity.prototype.getSubEntitiesByType = function(entityType) {
-	const vals = getMatchingValue(this._entitiesByType, entityType);
-	return vals ? vals.slice() : [];
-};
+import Action from './Action.js';
+import Entity from './Entity.js';
+import Field from './Field.js';
+import Link from './Link.js';
 
 export {
 	Entity,
 	Action,
-	Link
+	Link,
+	Field
```

Comment:
Mentioned in the description, but there is no `export default` from this library anymore. Thoughts/opinions? I'm fine to keep `export default Entity` perhaps, although in all honesty default exports in cases like this I've found to be a bit annoying in my experience (specifically around testing).

Reply:
I think this can be simplified to:

```javascript
export Action from './Action.js';
export Entity from './Entity.js';
export Field from './Field.js';
export Link from './Link.js';
```